sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py

```diff
@@ -425,108 +424,77 @@ async def negotiate(self):
         self._connect_timeout = connect_timeout or TIMEOUT_INTERVAL
         self._custom_endpoint = kwargs.get("custom_endpoint")
-        self.host, self.port = to_host_port(host, port)
+        self.host = host
         self.ws = None
-        self.session = None
         self._http_proxy = kwargs.get("http_proxy", None)
-        self.connected = False
 
     async def connect(self):
-        username, password = None, None
-        http_proxy_host, http_proxy_port = None, None
-        http_proxy_auth = None
-
+        http_proxy_host, http_proxy_port, http_proxy_auth = None, None, None
         if self._http_proxy:
             http_proxy_host = self._http_proxy["proxy_hostname"]
             http_proxy_port = self._http_proxy["proxy_port"]
-            if http_proxy_host and http_proxy_port:
-                http_proxy_host = f"{http_proxy_host}:{http_proxy_port}"
             username = self._http_proxy.get("username", None)
             password = self._http_proxy.get("password", None)
-
-        try:
-            from aiohttp import ClientSession
-            from urllib.parse import urlsplit
-
             if username or password:
-                from aiohttp import BasicAuth
-
-                http_proxy_auth = BasicAuth(login=username, password=password)
-
-            self.session = ClientSession()
-            if self._custom_endpoint:
-                url = f"wss://{self._custom_endpoint}"
-            else:
-                url = f"wss://{self.host}"
-                parsed_url = urlsplit(url)
-                url = f"{parsed_url.scheme}://{parsed_url.netloc}:{self.port}{parsed_url.path}"
+                http_proxy_auth = (username, password)
+        try:
+            from websocket import create_connection
 
-            self.ws = await self.session.ws_connect(
-                url=url,
+            self.ws = create_connection(
+                url="wss://{}".format(self._custom_endpoint or self.host),
+                subprotocols=[AMQP_WS_SUBPROTOCOL],
                 timeout=self._connect_timeout,
-                protocols=[AMQP_WS_SUBPROTOCOL],
-                autoclose=False,
-                proxy=http_proxy_host,
-                proxy_auth=http_proxy_auth,
-                ssl=self.sslopts,
+                skip_utf8_validation=True,
+                sslopt=self.sslopts,
+                http_proxy_host=http_proxy_host,
+                http_proxy_port=http_proxy_port,
+                http_proxy_auth=http_proxy_auth,
             )
-            self.connected = True
-
         except ImportError:
-            raise ValueError(
-                "Please install aiohttp library to use websocket transport."
-            )
+            raise ValueError("Please install websocket-client library to use websocket transport.")
 
-    async def _read(self, n, buffer=None, **kwargs):  # pylint: disable=unused-argument
+    async def _read(self, n, buffer=None, **kwargs):  # pylint: disable=unused-arguments
```

Comment:
might be a typo and need to be 
```suggestion
    async def _read(self, n, buffer=None, **kwargs):  # pylint: disable=unused-argument
```

Reply:
not sure if this will be a lot of work, but would you be able to compare with the sync transport/websocket to see if there were changes that need to be copied over here? I think I may have added some mypy/pylint things?

Reply:
not a lot of work, I just brought in the old `WebSocketTransportAsync` and the current portions untouched. Any re-shuffling will keep all the pylint/mypy work. Will take a look at the `WebSocketTransportAsync` though